### ISSUE TITLE
The Audio share and Audio setup buttons should not be toggle buttons

### ISSUE THREAD

**Issue by DavidBailes** [audacity/audacity#3911]
### Bug description

The Audio share and Audio setup buttons are set to be toggle buttons, which they are not. This affects both sighted users and users of screen readers as described in the steps to reproduce.

### Steps to reproduce

Case A.
1. Click on one of these buttons

Case B.
1. With a screen reader running, tab to one of these buttons.

### Expected behavior

Case A.
The button is shown as being down only when it is actually pressed.

Case B.
The button is read as a normal button.

### Actual behavior

Case A.
The button is shown down for the whole time that the menu is open.

Case B.
After PR https://github.com/audacity/audacity/pull/3909 has be committed so that toggle buttons are read better, the button will be read as being not pressed, incorrectly indicating that it is a toggle button.

### Audacity Version

latest stable version (from audacityteam.org/download)

### Operating system

Windows

**Comment by Ryan-Miller-Audacity** [comment]
Confirmed the above. Unfortunately it does appear that this regression was introduced recently and was working correctly previously.

**Comment by crsib** [comment]
I don't think that this is a regression. I wonder why Audio Setup was made a toggle in a first place, probably some workaround. @ksoze95 probably you remember?

**Comment by DavidBailes** [comment]
Any chance that my pull request for this bug could be merged for 3.2.2?

**Comment by crsib** [comment]
This needs to be tested by @Ryan-Miller-Audacity and @dozzzzer before. As I've told previously - I don't know why there Audio Setup was defined as a toggle and what breaks without it.

**Comment by DavidBailes** [comment]
> As I've told previously - I don't know why there Audio Setup was defined as a toggle and what breaks without it.

Without them being toggle buttons, when interacting using the keyboard, when enter is pressed they are not shown briefly as being down, which they should be - maybe this was why they were set to be toggle buttons. 

This is an excerpt from my commit message in which I should have said that the issue only arises when the button is "clicked" using the keyboard - mouse interaction is unaffected:

"Note that with this fix the buttons are not shown as down at the point when they are clicked, which they ideally should be. However, the menu opening gives visual feedback that the button has been pressed, and there are a number of other "menu buttons" already in Audacity which have the same behaviour."

Also in the effects panel, the menu button to the right of each effect name is an AButton, which is not set as a toggle button.

Also

**Comment by DavidBailes** [comment]
@dozzzzer . Thanks.

**Comment by DavidBailes** [comment]
> The fix looks good to me, however, on Mac when the button is pressed, it still appears to be in the "pressed" state as if it was a toggle. The initial bug report was raised for Windows, so I'm not sure whether it's of any importance in this case.

I think it would be best if any issues on Macs were considered separately from this issue.

**Comment by ksoze95** [comment]
> I don't think that this is a regression. I wonder why Audio Setup was made a toggle in a first place, probably some workaround. @ksoze95 probably you remember?

I think it was to stop the AudioSetup button creating another context/popup menu whilst one was already being displayed.  Not sure if it matters, I seem to remember that this was also originally coded up using wxMenu instead of BasicMenu.

**Comment by DavidBailes** [comment]
> I think it was to stop the AudioSetup button creating another context/popup menu whilst one was already being displayed. Not sure if it matters, I seem to remember that this was also originally coded up using wxMenu instead of BasicMenu.

On Windows 10, Audacity 3.2.1, using a mouse to click the AudioSetup button when a menu is already being displayed,  does reopen the menu, whereas standard behaviour would be to close it.
My fix does not change that exisiting behaviour.

If a menu button with more standard behaviour is required, then maybe AButton should have another "mode" of being a menu button, rather than reusing the toggle button mode.
